### PR TITLE
feat: Get Terms and Conditions

### PR DESCRIPTION
**Pull request by aronhoyer**
Resolves AB#86234

### CODE REVIEW

File: GetTermsAndConditionsTrigger/index.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import { AzureFunction, Context, HttpRequest } from "@azure/functions"
+import type { RightsToUse } from "@equinor/data-marketplace-models"
+import axios, { AxiosError } from "axios"
+import { config } from "../config"
+import { htmlToPortableText } from "../lib/html_to_portable_text"
+
+const GetTermsAndConditionTrigger: AzureFunction = async function (context: Context, req: HttpRequest): Promise<void> {
+  try {
+    const { data: relationTypeIDs } = await axios.get(`${config.COLLIBRA_BASE_URL}/relationTypes`, {
```
Comment (nilsml) [inline GetTermsAndConditionsTrigger/index.ts:9]: I think I would consider pulling out all these calls into separate functions. Will make the main function leaner and easier to follow.

File: lib/collibra/rights_to_use_adapter.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { RightsToUse } from "@equinor/data-marketplace-models"
+import { htmlToPortableText } from "../html_to_portable_text"
+
+export const rightsToUseAdapter = (asset: Collibra.AssetWithAttributes): RightsToUse => {
+  const descriptionAttr = asset.attributes.find((attr) => attr.type.name.toLowerCase() === "description")
+
+  const authURLattr = asset.attributes.find((attr) => attr.type.name.toLowerCase() === "authorization url")
+  let authURL: RightsToUse["authURL"] = {
```
Comment (nilsml) [inline lib/collibra/rights_to_use_adapter.ts:8]: Let?? 😱

File: lib/html_to_portable_text/index.ts [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
+import type { PortableTextBlock } from "@portabletext/types"
 
 import { collibraRichText } from "./schemas/collibraRichText"
 import { rules } from "./rules"
 import { sanitizeHtml } from "../sanitize_html"
 import { htmlStringToDom } from "../html_string_to_dom"
 
-export const htmlToPortableText = (html: string) => {
+export const htmlToPortableText = (html: string): PortableTextBlock[] => {
   return htmlToBlocks(html, collibraRichText.get("content").fields.find((field) => field.name === "body").type, {
```
Comment (nilsml) [inline lib/html_to_portable_text/index.ts:10]: I know this is not part of the PR, but the return statement here seems unnecessary

**Review by nilsml — CHANGES_REQUESTED** [review]
Split up nicely now. A couple of minor things I would like you to fix, but overall good!

File: lib/html_to_portable_text/index.ts (hunk 3 above)
Comment (aronhoyer) [reply lib/html_to_portable_text/index.ts:10]: it is, but in my eyes, it increases legibility

Comment (nilsml) [reply lib/html_to_portable_text/index.ts:10]: I disagree. I think it would be more declarative and easier to read without the return.

**Review by nilsml — CHANGES_REQUESTED** [review]

**Review by nilsml — APPROVED** [review]
Good stuff!

**Pull request merged**